### PR TITLE
Move xsoar.pan.dev to firebase deployment

### PR DESCRIPTION
**Pull request by reutshal**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/42062

## Description
Moving the xsoar.pan.dev deployment from Netlify to Firebase

### CODE REVIEW

File: content-repo/post_github_comment.py [hunk 1]
```diff
@@ -96,22 +96,34 @@ def post_comment(netlify_deploy_file: str):
...
+    # handle Firebase deployment message
+    else:
+        with open(netlify_deploy_file, 'r') as f:
+            if matched_url := re.search("https://xsoar-pan-dev--pull-request-.*web.app", f.read()):
+                deploy_url = matched_url.group(0)
+            else:
+                raise ValueError("Can't post comment. Deployment was unsuccessful: \n" + f.read())
+
     # preview message
     message = "# Preview Site Available\n\n" \
```
Comment (yaakovi) [inline content-repo/post_github_comment.py:115]: add to the title the site hosting (netlify/firebase) at least until we finish the migration and remove the double comments

**Comment by glicht** [comment]
@reutshal need to migrate the redirects we have defined in: https://github.com/demisto/content-docs/blob/master/static/_redirects

See firebase docs on how to define redirects: https://firebase.google.com/docs/hosting/full-config#redirects

For example this link works: https://pull-request-817--demisto-content-docs.netlify.app/start-contributing
And this doesn't: https://xsoar-pan-dev--pull-request-817-rpjbf3v3.web.app/start-contributing

File: .circleci/config.yml [hunk 2]
```diff
@@ -106,24 +106,37 @@ jobs:
...
+
+              # Firebase production deployment
+              firebase deploy --only hosting > deploy-info-firebase.txt
             else
               # Get the PR number from the CIRCLE_PULL_REQUEST env variable which is in the form of:
               # https://github.com/demisto/content-docs/pull/335
               DEPLOG_ARG="--alias=pull-request-${CIRCLE_PULL_REQUEST##*/}"
+
+              # Firebase preview deployment
+              firebase hosting:channel:deploy "pull-request-${CIRCLE_PULL_REQUEST##*/}" --only xsoar.pan.dev > deploy-info-firebase.txt
```
Comment (glicht) [inline .circleci/config.yml:124]: Add after this: `cat deploy-info-firebase.json | jq` so the  output is clear as we do with netlify

**Comment by glicht** [comment]
Also need to create a workflow file which we use for forked PRs. 

See: https://github.com/demisto/content-docs/blob/master/.github/workflows/forked-pr-netlify-deploy.yml

You can rename the file to forked-pr-deploy.yml and add support for both netlify and firebase.

File: .circleci/config.yml (hunk 2 above)
Comment (reutshal) [reply .circleci/config.yml:124]: Did that at first, but the output of the command is txt and not json

Comment (glicht) [reply .circleci/config.yml:124]: So just include the text

File: .circleci/config.yml [hunk 3]
```diff
@@ -106,27 +106,39 @@ jobs:
...
             fi
 
             # Netlify Deployment
             netlify deploy -m "CircleCI build: $CIRCLE_BRANCH" -d build $DEPLOG_ARG --json > deploy-info.json
             cat deploy-info.json | jq
+      - run:
+          name: Deploy to Firebase
+          no_output_timeout: 30m
+          command: |
+            npm install firebase-tools
```
Comment (yaakovi) [inline .circleci/config.yml:128]: add a print to the firebase-tools version like we do for netlify.

**Review by glicht — APPROVED** [review]
Looks good. After merging please open a forked PR (external user) and test.

**Pull request merged**